lib/createGltf.js

```diff
@@ -13,335 +16,902 @@ module.exports = createGltf;
+    var count = array.length;
+    var minMax = array.getMinMax(1);
+
+    var accessor = {
+        name : name,
+        componentType : componentType,
+        count : count,
+        min : minMax.min,
+        max : minMax.max,
+        type : 'SCALAR'
+    };
+
+    var accessorIndex = gltf.accessors.length;
+    gltf.accessors.push(accessor);
+    return accessorIndex;
+}
+
+function requiresUint32Indices(nodes) {
+    var nodesLength = nodes.length;
+    for (var i = 0; i < nodesLength; ++i) {
+        var meshes = nodes[i].meshes;
+        var meshesLength = meshes.length;
+        for (var j = 0; j < meshesLength; ++j) {
+            // Reserve the 65535 index for primitive restart
+            var vertexCount = meshes[j].positions.length / 3;
+            if (vertexCount > 65534) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+function addMesh(gltf, materials, images, bufferState, uint32Indices, mesh, options) {
+    var hasPositions = mesh.positions.length > 0;
+    var hasNormals = mesh.normals.length > 0;
+    var hasUVs = mesh.uvs.length > 0;
+
+    var accessorIndex;
+    var attributes = {};
+    if (hasPositions) {
+        accessorIndex = addVertexAttribute(gltf, mesh.positions, 3, mesh.name + '_positions');
+        attributes.POSITION = accessorIndex;
+        bufferState.positionBuffers.push(mesh.positions.toFloatBuffer());
+        bufferState.positionAccessors.push(accessorIndex);
+    }
+    if (hasNormals) {
+        accessorIndex = addVertexAttribute(gltf, mesh.normals, 3, mesh.name + '_normals');
+        attributes.NORMAL = accessorIndex;
+        bufferState.normalBuffers.push(mesh.normals.toFloatBuffer());
+        bufferState.normalAccessors.push(accessorIndex);
+    }
+    if (hasUVs) {
+        accessorIndex = addVertexAttribute(gltf, mesh.uvs, 2, mesh.name + '_texcoords');
+        attributes.TEXCOORD_0 = accessorIndex;
+        bufferState.uvBuffers.push(mesh.uvs.toFloatBuffer());
+        bufferState.uvAccessors.push(accessorIndex);
+    }
+
+    // Unload resources
+    mesh.positions = undefined;
+    mesh.normals = undefined;
+    mesh.uvs = undefined;
+
+    var gltfPrimitives = [];
+    var primitives = mesh.primitives;
+    var primitivesLength = primitives.length;
+    for (var i = 0; i < primitivesLength; ++i) {
+        var primitive = primitives[i];
+        var indexAccessorIndex = addIndexArray(gltf, primitive.indices, uint32Indices, mesh.name + '_' + i + '_indices');
+        var indexBuffer = uint32Indices ? primitive.indices.toUint32Buffer() : primitive.indices.toUint16Buffer();
+        bufferState.indexBuffers.push(indexBuffer);
+        bufferState.indexAccessors.push(indexAccessorIndex);
+
+        primitive.indices = undefined; // Unload resources
+
+        var materialIndex = getMaterial(gltf, materials, images, primitive.material, hasNormals, options);
+
+        gltfPrimitives.push({
+            attributes : attributes,
```

Comment:
Is it possible in gltf 2.0 for different primitives in the same mesh to have differing numbers of attributes?